tests/feeder/wrapperTests.cpp

```diff
@@ -0,0 +1,175 @@
+    ASSERT_EQ(children[2]->id(), 435);
+  } else if (ext == "json") {
+    std::vector<JSONNode> children;
+    node.getChildren(children);
+    ASSERT_EQ(children[0].id(), 217);
+    ASSERT_EQ(children[1].id(), 430);
+    ASSERT_EQ(children[2].id(), 435);
+  }
+}
+
+TEST_F(WrapperNodeTest, RemoveTest) {
+  // SetUp("tests/data/chakra.0.et");
+  SetUp("tests/data/small_chakra.0.json");
+  node.lookupNode(216);
+  ASSERT_EQ(node.getNodeID(), 216);
+  node.removeNode(216);
+  freopen("/dev/null", "w", stderr);
+  try {
+    node.lookupNode(216);
+    ASSERT_TRUE(false) << "node should be removed \n";
+  } catch (const std::exception& e) {
+    // this is the desired behaviour
+  }
+  freopen("/dev/tty", "w", stderr);
+}
+
+TEST_F(WrapperNodeTest, RemoveAndGetNextTest) {
+  // SetUp("tests/data/chakra.0.et");
+  SetUp("tests/data/small_chakra.0.json");
+  node.lookupNode(216);
+  ASSERT_EQ(node.getNodeID(), 216);
+  node.removeNode(216);
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 216);
+}
+
+TEST_F(WrapperNodeTest, FreeChildrenTest) {
+  // SetUp("tests/data/chakra.0.et");
+  SetUp("tests/data/small_chakra.0.json");
+  node.lookupNode(216);
+  ASSERT_EQ(node.getNodeID(), 216);
+  node.freeChildrenNodes(216);
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 216);
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 217);
+}
+
+TEST_F(WrapperNodeTest, HasNodesToIssueTest) {
+  // SetUp("tests/data/chakra.0.et");
+  SetUp("tests/data/small_chakra.0.json");
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 216);
+  ASSERT_TRUE(node.hasNodesToIssue());
+  node.removeNode(5);
+  ASSERT_TRUE(node.hasNodesToIssue());
+}
+
+TEST_F(WrapperNodeTest, PushBackIssuableNodeTest) {
+  // SetUp("tests/data/chakra.0.et");
+  SetUp("tests/data/small_chakra.0.json");
+  node.pushBackIssuableNode(217);
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 216);
+  node.getNextIssuableNode();
+  ASSERT_EQ(node.getNodeID(), 217);
+}
+
+TEST_F(WrapperNodeTest, AddNodeTest) {
+  // std::string filename = "tests/data/chakra.0.et";
+  std::string filename = "tests/data/small_chakra.0.json";
+  std::string ext = filename.substr(filename.find_last_of(".") + 1);
+  SetUp(filename);
+  if (ext == "et") {
+    std::shared_ptr<Chakra::ETFeederNode> pnode1;
+    node.lookupNode(216);
+    pnode1 = node.getProtobufNode();
+    node.removeNode(216);
+    node.addNode(pnode1);
+    std::shared_ptr<Chakra::ETFeederNode> pnode2;
+    node.lookupNode(216);
+    pnode2 = node.getProtobufNode();
+    ASSERT_EQ(pnode2->id(), 216);
+  } else if (ext == "json") {
+    JSON jnode1;
+    node.lookupNode(216);
+    jnode1 = node.getJSONNode();
+    node.removeNode(216);
+    node.addNode(jnode1);
+    JSONNode jnode2;
+    node.lookupNode(216);
+    jnode2 = node.getJSONNode();
+    ASSERT_EQ(jnode2.id(), 216);
+  }
+}
+
+TEST_F(WrapperNodeTest, NodeGetChildrenTest) {
+  // std::string filename = "tests/data/chakra.0.et";
```

Comment:
Please don't comment out codes.

Reply:
I have two input files for testing: `tests/data/chakra.0.json` and `tests/data/small_chakra.0.json`. small_chakra.0.json is a pruned version of the former for quick testing. I left a comment there to indicate the availability of the full dataset.